src/transactions/test/SorobanTxTestUtils.h

```diff
@@ -214,28 +214,34 @@ class ContractStorageInvocationTest : public WasmContractInvocationTest
     {
     }
 
-    void put(std::string const& key, ContractDataDurability type, uint64_t val,
-             bool expectSuccess = true);
-    void putWithFootprint(std::string const& key, ContractDataDurability type,
-                          uint64_t val, xdr::xvector<LedgerKey> const& readOnly,
-                          xdr::xvector<LedgerKey> const& readWrite,
-                          bool expectSuccess, uint32_t writeBytes = 1000,
-                          uint32_t refundableFee = 40'000);
-
-    uint64_t get(std::string const& key, ContractDataDurability type,
-                 bool expectSuccess = true);
+    void
+    put(std::string const& key, ContractDataDurability type, uint64_t val,
+        InvokeHostFunctionResultCode result = INVOKE_HOST_FUNCTION_SUCCESS);
+    void putWithFootprint(
+        std::string const& key, ContractDataDurability type, uint64_t val,
+        xdr::xvector<LedgerKey> const& readOnly,
+        xdr::xvector<LedgerKey> const& readWrite,
+        InvokeHostFunctionResultCode result = INVOKE_HOST_FUNCTION_SUCCESS,
```

Comment:
This probably should rather return the `result` (and `invokeTx` should return `bool`), so that more checks actually happen in the test code and allow to attribute failures to the actual lines where tx has failed. Probably not necessary for this PR though...